src/Nethermind/Nethermind.BeaconNode.OApi/Controllers/ValidatorAttestationController.cs

```diff
@@ -0,0 +1,74 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Nethermind.Core2;
+using Nethermind.Core2.Api;
+using Nethermind.Core2.Containers;
+using Nethermind.Core2.Crypto;
+using Nethermind.Core2.Types;
+using Nethermind.Logging.Microsoft;
+
+namespace Nethermind.BeaconNode.OApi.Controllers
+{
+    [ApiController]
+    [Route("/validator/attestation")]
+    public class ValidatorAttestationController : ControllerBase
+    {
+        private readonly IBeaconNodeApi _beaconNode;
+        private readonly ILogger _logger;
+
+        public ValidatorAttestationController(ILogger<ValidatorAttestationController> logger, IBeaconNodeApi beaconNode)
+        {
+            _logger = logger;
+            _beaconNode = beaconNode;
+        }
+
+        /// <summary>Produce an attestation, without signature.</summary>
+        /// <remarks>
+        /// <para>
+        /// Requests that the beacon node produce an Attestation, with a blank signature field, which the validator will then sign.
+        /// </para>
+        /// </remarks>
+        /// <param name="validator_pubkey">Uniquely identifying which validator this attestation is to be produced for.</param>
+        /// <param name="poc_bit">The proof-of-custody bit that is to be reported by the requesting validator. This bit will be inserted into the appropriate location in the returned `Attestation`.</param>
+        /// <param name="slot">The slot for which the attestation should be proposed.</param>
+        /// <param name="index">The index number for which the attestation is to be proposed.</param>
+        [HttpGet]
```

Comment:
this seems helpful - I think you could help us to fix the routing for the API - this is why I was discussing with Sly that you could look at the OAPI for Eth2

Reply:
> there should be some route for this endpoint 

I have one controller per operation, including separate controllers for the GET and POST of attestation. Note it is using explicit API endpoints (not generated routing), i.e. the endpoint is just explicit "/validator/attestation" , *not* "{controller}/{method}"; the name of the method is irrelevant.

The OpenAPI specification is defined in the Eth 2.0 specifications, see the apis/validator/beacon-node-oapi-yaml specs in the source code.

The 0.10.1 version of the spec says the values are passed as parameters, i.e.

/validator/attestation?validator_pubkey=0xAB...&poc_bit=1&slot=27&shard=3

Caveats: The 0.10.1 version of the OpenAPI in the spec was not updated with the rest of the spec, so contains some properties that were removed from the spec. In particular, shard was removed from phase 0 and replaced with (committee) index, but OAPI was not updated.

Thus the actual implementation is:
/validator/attestation?validator_pubkey=0xAB...&poc_bit=1&slot=27&index=3

Because the processing code (which was updated in the spec) actually uses index, not shard.

There have been further changes in the spec since then, plus there is ongoing discussion/changes about the API specs, which are sort of an extra add on to the main spec.